src/utils/getLatestRevision.ts

```diff
@@ -0,0 +1,17 @@
+import type {
+  ListSecretResult,
+  SecretRevision,
+} from "@canonical/jujulib/dist/api/facades/secrets/SecretsV2";
+
+const getLatestRevision = (secret: ListSecretResult | null | undefined) => {
+  if (!secret) return -1;
+  return (secret?.revisions ?? []).reduce(
+    (result: number, { revision }: SecretRevision) => {
+      if (revision > result) result = revision;
+      return result;
```

Comment:
I think you can do this, which is slightly more terse, but both work:

```suggestion
      => Math.max(revision, result)
```

Reply:
Smart. Done